### PR TITLE
CAMEL-18620: Add redpanda container

### PR DESCRIPTION
**Pull request by Croway**

### CODE REVIEW

**Review by oscerd — CHANGES_REQUESTED** [review]
Minor findings.

File: test-infra/camel-test-infra-kafka/src/test/java/org/apache/camel/test/infra/kafka/services/RedpandaTransactionsEnabledContainer.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
+package org.apache.camel.test.infra.kafka.services;
```
Comment (orpiske) [inline test-infra/camel-test-infra-kafka/src/test/java/org/apache/camel/test/infra/kafka/services/RedpandaTransactionsEnabledContainer.java:17]: This file is missing the license header.

File: test-infra/camel-test-infra-kafka/src/test/java/org/apache/camel/test/infra/kafka/services/RedpandaService.java [hunk 2]
```diff
@@ -0,0 +1,70 @@
+package org.apache.camel.test.infra.kafka.services;
```
Comment (orpiske) [inline test-infra/camel-test-infra-kafka/src/test/java/org/apache/camel/test/infra/kafka/services/RedpandaService.java:17]: This file is missing the license header.

File: test-infra/camel-test-infra-kafka/src/test/java/org/apache/camel/test/infra/kafka/services/RedpandaTransactionsEnabledContainer.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (Croway) [reply test-infra/camel-test-infra-kafka/src/test/java/org/apache/camel/test/infra/kafka/services/RedpandaTransactionsEnabledContainer.java:17]: ops, thanks, force pushed

**Review by orpiske — CHANGES_REQUESTED** [review]
Thanks for your contribution. The code LGTM, but it is missing the license header on the top of the files.



Can you adjust that, please?

**Review by oscerd — APPROVED** [review]

**Review by orpiske — APPROVED** [review]
LGTM. Let's keep them turned off by default for now and then we can reevaluate to see how things progress.

Thanks!

**Review by davsclaus — APPROVED** [review]

**Comment by oscerd** [comment]
Restarted.

**Comment by orpiske** [comment]
> The following test failed `org.apache.camel.processor.PooledExchangeTest.testSameExchange` in camel-core, but it is not part of the PR.

This usually means that you need to rebase the code. Unfortunately we don't have a way to force PRs to do auto-rebase.

**Comment by Croway** [comment]
> This usually means that you need to rebase the code. Unfortunately we don't have a way to force PRs to do auto-rebase.

I see, rebased.

**Comment by emaxerrno** [comment]
@Croway - happy to help debug the failed test. we have a set of transaction improvement landing on Nov 15th in 2 weeks. Let me know if you want us to chase that failed yet. happy to collab on it.

**Comment by Croway** [comment]
Hello @emaxerrno, as you can see from this comment https://issues.apache.org/jira/browse/CAMEL-18620?focusedCommentId=17619669&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-17619669 there are some failures, I've enabled the transaction via

`/usr/bin/rpk redpanda config set redpanda.enable_transactions true\n`

and this is working as expected, but I didn't spend much time on the remaining failures. 

I'll try again to execute the tests once a new image with transaction improvement is ready, I'll monitor `docker.redpanda.com/vectorized/redpanda` in mid november.

**Comment by emaxerrno** [comment]
@Croway let me chase. will keep you posted.

**Pull request merged**